### PR TITLE
Feature circuit breaker

### PR DESCRIPTION
**Pull request by BloodyHell**
The dataflow will now not be broken anymore when the database is offline.

Also, added a circuit breaker to the database connection. So when the database is responding slowly, the calls will automatically go to the RIOT Api.

### CODE REVIEW

File: src/main/scala/com/leagueprojecto/api/services/couchdb/DatabaseService.scala [hunk 1]
```diff
@@ -42,11 +43,6 @@ class DatabaseService extends Actor with ActorLogging {
   val summonerDb = couch.db("summoner-db", summonerMapping)
   val matchesDb = couch.db("matches-db", matchMapping)
 
-  val breaker =
-    new CircuitBreaker(context.system.scheduler, maxFailures = 5, callTimeout = 10.seconds, resetTimeout = 1.minute)
-      .onOpen(whenOpen())
-
-
   private[this] def whenOpen(): Unit = {
     println("Circuit breaker open")
```
Comment (MikeMolIS) [inline src/main/scala/com/leagueprojecto/api/services/couchdb/DatabaseService.scala]: Don't use println

File: src/it/scala/com/leagueprojecto/it/EndpointTest.scala [hunk 2]
```diff
@@ -14,7 +14,7 @@ abstract class EndpointTest extends FlatSpec with Matchers with GivenWhenThen wi
   override val logger: LoggingAdapter = system.log
 
   // Services
-  val matchHistoryService: ActorRef = system.actorOf(MatchHistoryManager.props)
+  //val matchHistoryService: ActorRef = system.actorOf(MatchHistoryManager.props)
```
Comment (MikeMolIS) [inline src/it/scala/com/leagueprojecto/it/EndpointTest.scala]: remove commented lines

File: src/main/scala/com/leagueprojecto/api/services/couchdb/DatabaseService.scala [hunk 3]
```diff
@@ -38,38 +43,51 @@ class DatabaseService extends Actor with ActorLogging {
   val summonerDb = couch.db("summoner-db", summonerMapping)
   val matchesDb = couch.db("matches-db", matchMapping)
 
+  private[this] def whenOpen(): Unit = {
+    println("Circuit breaker open")
```
Comment (MikeMolIS) [inline src/main/scala/com/leagueprojecto/api/services/couchdb/DatabaseService.scala]: Don't use println, but use a logger instead

File: src/main/scala/com/leagueprojecto/api/services/couchdb/DatabaseService.scala [hunk 4]
```diff
@@ -1,29 +1,34 @@
...
+
 import com.ibm.couchdb.Res.Error
-import com.ibm.couchdb.{CouchException, TypeMapping, CouchDb}
+import com.ibm.couchdb.{CouchDb, CouchException, TypeMapping}
 import com.leagueprojecto.api.domain.{MatchDetail, Summoner}
 import org.http4s.Status.NotFound
 
-import scalaz.{\/-, -\/}
+import scala.util.{Failure, Success, Try}
+import scalaz.{-\/, \/, \/-}
```
Comment (MartinKanters) [inline src/main/scala/com/leagueprojecto/api/services/couchdb/DatabaseService.scala]: I like this line

File: src/main/scala/com/leagueprojecto/api/services/couchdb/DatabaseService.scala [hunk 5]
```diff
@@ -42,54 +47,69 @@ class DatabaseService extends Actor with ActorLogging {
...
         case \/-(matchesDocs) =>
           log.info(s"Yay, got ${matchesDocs.getDocsData.size} matches from Db")
           sender() ! MatchesResult(matchesDocs.getDocsData)
+        case -\/(CouchException(e: Error)) if e.status == NotFound =>
+          log.info(s"No matches found ($matchIds) from Db")
+          sender() ! NoResult
         case -\/(e) =>
           log.error(s"Error retrieving matches ($matchIds) from Db")
-          e.printStackTrace()
+          log.error(s"${e.getMessage}")
```
Comment (MartinKanters) [inline src/main/scala/com/leagueprojecto/api/services/couchdb/DatabaseService.scala]: Hmm, can't we just log the whole exception? 
Aside from that we should try to keep the exception handling the same way in all those methods.

**Pull request merged**